contracts/svm/programs/express_relay/src/lib.rs

```diff
@@ -309,3 +463,94 @@ pub struct WithdrawFees<'info> {
     #[account(mut, seeds = [SEED_METADATA], bump, has_one = admin)]
     pub express_relay_metadata: Account<'info, ExpressRelayMetadata>,
 }
+
+#[derive(AnchorSerialize, AnchorDeserialize, Eq, PartialEq, Clone, Copy, Debug)]
+pub enum FeeToken {
+    Input,
+    Output,
+}
+
+/// For all swap instructions and contexts, input and output are defined with respect to the searcher
+/// So mint_input refers to the token that the searcher provides to the trader
+/// mint_output refers to the token that the searcher receives from the trader
+/// This choice is made to minimize confusion for the searchers, who are more likely to parse the program
+#[derive(AnchorSerialize, AnchorDeserialize, Eq, PartialEq, Clone, Copy, Debug)]
+pub struct SwapArgs {
+    pub amount_input:  u64,
+    pub amount_output: u64,
+    // The referral fee is specified in parts per million (e.g. 100 = 1 bp)
+    pub referral_fee:  u64,
+    // Token in which the fees will be paid
+    pub fee_token:     FeeToken,
+}
+
+#[derive(Accounts)]
+#[instruction(data: Box<SwapArgs>)]
+pub struct Swap<'info> {
+    /// Express relay configuration
+    #[account(seeds = [SEED_METADATA], bump)]
+    pub express_relay_metadata: Account<'info, ExpressRelayMetadata>,
```

Comment:
slight nit, can we keep a similar order to what we have in other instructions? signers usually first, followed by other accounts, `express_relay_metadata` usually near the end iirc

Reply:
moved `express_relay_metadata` to the end and both signers to the top